pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,207 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
                     nextlevel[w] = 1
         level += 1
     return paths
+
+
+def _find_directed_parents(G, node):
+    """Finds the parents of a node in the directed and the bidirected subgraphs.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node : node label
+        The node for which we have to find the parents.
+
+    Returns
+    -------
+    out : set
+        The parents of the provided node.
+    """
+
+    if not isinstance(G, CPDAG):
```

Comment:
Not sure why this check is needed.

Reply:
Because a CPDAG does not have a bidirected subgraph. At least not in the class defintion.

Reply:
Let's assume the graph is a mixed edge graph. If so, it may or may not have any of these edge subgraphs.